lib/console2/searchui.js

```diff
@@ -0,0 +1,137 @@
+'use babel'
+/** @jsx etch.dom */
+
+import etch from 'etch'
+import { Raw, Button, toView } from '../util/etch.js'
+import { CompositeDisposable, Emitter, TextEditor, TextBuffer, Disposable, MarkerLayer, Range } from 'atom'
+import { Terminal } from 'xterm'
+import * as fit from 'xterm/lib/addons/fit/fit'
+import * as webLinks from 'xterm/lib/addons/webLinks/webLinks'
+import * as winptyCompat from 'xterm/lib/addons/winptyCompat/winptyCompat'
+import * as search from 'xterm/lib/addons/search/search'
+import TerminalElement from './view'
+import PaneItem from '../util/pane-item'
+import ResizeDetector from 'element-resize-detector'
+import { debounce, throttle } from 'underscore-plus'
+import { closest } from './helpers'
+import { openExternal } from 'shell'
+
+export default class TerminalSearchUI {
```

Comment:
nit: This class actually does the UI /and/ the search functionality. Would like `SearchBar` or `SearchPanel` or `SearchBox` be a reasonable name instead?

-- oh, wait, i guess "UI" doesn't necessarily mean "html-only"... Maybe ignore me. That said, when i first glanced through this PR i scrolled past this file expecting it just to be the HTML/CSS stuff, and expecting the functionality to be implemented somewhere else, so idk if others might have the same thought.